AmazonChimeSDK/AmazonChimeSDK/internal/video/VideoFrameResender.swift

```diff
@@ -13,88 +13,98 @@ import CoreMedia
+        let resendTimeInterval:CMTime = CMTime(value: CMTimeValue(minFrameInterval),
+                                         timescale: CMTimeScale(Constants.millisecondsPerSecond))
+        
+        self.resendTimer.setEventHandler(handler: { [weak self] in
             guard let `self` = self else {
-                timer.cancel()
+                self?.resendTimer.cancel()
+                self?.logger.error(msg: "Unable to resend video frame, VideoFrameResender is unavailable. Thread: \(Thread.current)")
                 return
             }
-
+            
+            self.logger.info(msg: "Checking if there is pending frame for resending. Thread: \(Thread.current)")
+            lock.lock()
             guard let lastSendTimestamp = self.lastSendTimestamp,
                   let lastVideoFrame = self.lastVideoFrame else { return }
+            lock.unlock()
 
+            self.logger.info(msg: "Checking the time elapsed for resending. Thread: \(Thread.current)")
             let currentTimestamp = CMClockGetTime(CMClockGetHostTimeClock())
             let delta = CMTimeSubtract(currentTimestamp, lastSendTimestamp)
 
             // Resend the last input frame if there is no new input frame after resendTimeInterval
-            if delta > self.resendTimeInterval {
+            if delta > resendTimeInterval {
+                self.logger.info(msg: "Creating new video frame for resending. Thread: \(Thread.current)")
                 // Update the timestamp so it's not dropped by downstream as a duplicate
                 let lastVideoFrameTime = CMTimeMake(value: lastVideoFrame.timestampNs,
                                                     timescale: Int32(Constants.nanosecondsPerSecond))
                 let newVideoFrame = VideoFrame(timestampNs: Int64(CMTimeAdd(lastVideoFrameTime, delta).seconds
                                                                     * Double(Constants.nanosecondsPerSecond)),
                                                rotation: lastVideoFrame.rotation,
                                                buffer: lastVideoFrame.buffer)
-
-                // Cancel the current timer so that new frames will kick it off again
-                self.resendTimer?.cancel()
-                self.resendTimer = nil
-
-                self.resendFrameHandler(newVideoFrame)
-            } else {
-                // Reset resending schedule if there is an input frame between internals
-                let remainingSeconds = self.resendTimeInterval.seconds - delta.seconds
-                let deadline = DispatchTime.now() + DispatchTimeInterval.milliseconds(Int(remainingSeconds *
-                                                           Double(Constants.millisecondsPerSecond)))
-                self.resendTimer?.schedule(deadline: deadline, leeway: self.resendScheduleLeewayMs)
+                self.logger.info(msg: "Resending last frame. Thread: \(Thread.current)")
+                resendFrameHandler(newVideoFrame)
```

Comment:
One thing worth mentioning, is that this technically requires the user of the class to ensure this callback is thread safe (which we don't currently do). We should at least put a comment in the `init` above this param though.

Reply:
Will do!